desk/app/heap.hoon

```diff
@@ -951,7 +950,11 @@
                 (flatten q.content.heart)
             ==
           ~
-        =.  cor  (emit (pass-hark & & yarn))
+        =/  am-op-author  =(author.curio.u.op our.bowl)
+        =/  am-author  =(author.heart our.bowl)
+        ?:  |(&(!am-author in-replies) &(am-op-author !am-author))
+          =.  cor  (emit (pass-hark & & yarn))
+        he-core
```

Comment:
the reason is that doing `=.` makes a state change and then returns the value following using that updated subject so `=.  cor  (emit (pass-hark & & yarn))` does the state change, but you have to return something.

```suggestion
        =?  cor  |(&(!am-author in-replies) &(am-op-author !am-author))
          (emit (pass-hark & & yarn))
        he-core
```

this will do the state change conditionally and then always return the core